src/dp.rs

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Differential privacy (DP) primitives.
+use crate::field::FieldElement;
+use crate::vdaf::prg::Seed;
+use std::fmt::Debug;
+
+// TODO(tholop): We could also use pairs of integers, or even
+// floats to store the DiscreteGaussian DP parameters?
+// Using BigInts might make it harder to communicate `DiscreteGaussian` to the Collector.
+use num_rational::BigRational;
+
+/// DP-related errors.
+#[derive(Debug, thiserror::Error)]
+pub enum DpError {
+    /// Error while sampling from a specific noise distribution
+    #[error("sampling error")]
+    SamplingError,
+}
+
+/// Trait for a noise distribution over a (vector space over a) finite field.
+/// Useful for differential privacy.
+/// Inherited by specific mechanism instantiations such as `[DiscreteGaussian]`
+/// that will enforce the VDAF-dependent DP guarantees.
+pub trait Dist: Clone + Debug {
+    /// Sample `len` elements from the distribution over `F`.
+    /// The randomness comes from a Prg seeded by `seed`.
+    /// TODO(tholop): do we need a generic SEED_SIZE here, or 16 is fine?
+    fn sample<F: FieldElement, const SEED_SIZE: usize>(
+        &self,
+        seed: Seed<SEED_SIZE>,
+        len: usize,
+    ) -> Result<Vec<F>, DpError>;
+
+    // TODO(tholop): do we need another method to send the distribution parameters
+    // to the Collector?
+}
+
+/// Discrete Gaussian noise distribution with mean 0 and standard deviation `sigma`.
+/// `sigma` can be computed as a function of DP parameters `epsilon`, `delta`
+/// and `sensitivity`. `sensitivity` depends on the VDAF and potentially the number of measurements.
+/// `sigma` is sufficient to characterize the distribution, but other parameters
+/// are kept for interpretability.
```

Comment:
Style: In this paragraph you refer to variables that are not exposed in the public API. I'd suggest moving this discussion to the constructor (`::new()`) below.

Reply:
I thought these variables were exposed since they are fields of a `pub struct`. Is that not the case?

Reply:
No -- the structure itself is public so `DiscreteGaussian` values can be used outside this module, but its fields are private and not visible outside this module unless each is marked `pub`. See [docs](https://doc.rust-lang.org/rust-by-example/mod/struct_visibility.html). n.b. visibility of enum variants does not work the same way, which could be the source of your confusion?

Reply:
My bad, thanks!